### PR TITLE
Imorove prime calc code

### PR DESCRIPTION
**Pull request by NAVNAV221**
prime code return "This is prime number" for any number.

### CODE REVIEW

File: primary.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	for (int i = 2; i <= sqrt(number); i++)
+	{
+		if (number % i == 0)
+			return false;
+	}
+	return true;
+}
+
+int main(int argc, char *argv[]){
+	int num = *argv[1];
```
Comment (sean-wolmer-cyera) [inline primary.c]: This is an implicit cast that causes undefined behavior later on

File: primary.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	num = convert(num);
+	if (isPrime(num)){
+	printf("This is prime number");
+	}
+	else{
+		printf("This is prime number");
```
Comment (sean-wolmer-cyera) [inline primary.c]: You print that the number is prime if it isn't prime...

File: primary.c (hunk 1 above)
Comment (sean-wolmer-cyera) [inline primary.c]: There is no check that the user supplied the number in the argv

File: primary.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#include<stdio.h>
+#include<math.h>
+#include<stdbool.h>
+
+int convert(int number){
```
Comment (sean-wolmer-cyera) [inline primary.c]: You get a number and iterate through every digit just to rebuild the same number.
This doesn't convert anything. 
If you want to know how to convert a string representing a number, ru is the "man atoi"

File: primary.c (hunk 1 above)
Comment (NAVNAV221) [reply primary.c]: Yes, I need to work on the parameter that I get from argv like char.

File: primary.c (hunk 2 above)
Comment (NAVNAV221) [reply primary.c]: YES I forgot

File: primary.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#include<stdio.h>
+#include<math.h>
+#include<stdbool.h>
+#include<string.h>
+#include <stdlib.h>
+
+int convert(int number){
```
Comment (sean-wolmer-cyera) [inline primary.c]: Remove this function

File: primary.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	for (int i = 2; i <= sqrt(number); i++)
+	{
+		if (number % i == 0)
+			return false;
+	}
+	return true;
+}
+
+int main(int argc, char *argv[]){
+	int num = atoi(argv[1]);
```
Comment (sean-wolmer-cyera) [inline primary.c]: You don't check argc

File: primary.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	int check_num = num;
+	while (check_num != 0 && (check_num % 10) <= 1 )
+	{
+ 	  	check_num /= 10;
+	}
+	if ((check_num % 10) > 1 )
+	{	
+		printf("Not binary number!");
```
Comment (sean-wolmer-cyera) [inline primary.c]: Why do you convert to binary?

File: primary.c (hunk 3 above)
Comment (NAVNAV221) [reply primary.c]: I convert binary number to decimal number, for example:
**input from user:** 0111. **convert:** 7

File: primary.c (hunk 5 above)
Comment (NAVNAV221) [reply primary.c]: Yes, you right I need to add that one.

File: primary.c (hunk 6 above)
Comment (NAVNAV221) [reply primary.c]: If I get from argv unbinary number (Ex: 7 instad of 0111), so I need to alert that. You required that the code act only for binary numbers.

**Review by montito99 — APPROVED** [review]

**Pull request merged**